### PR TITLE
Gunpoint now blocks bumps, adds examine text and can be broken by shoving

### PR DESCRIPTION
**Pull request by larentoun**
## About The Pull Request
Lesser version of https://github.com/tgstation/tgstation/pull/75226

Changes a few things with bumping which could lead to cheesing a charged shot if the shooter has an ally to bump the target. Also adds examine text to know what's happening.
Also shoving now can be used to break gunpoint, since having immovable mobs can be troublesome in some situations

## Why It's Good For The Game
Grabs from the target no longer counter gunpoint;
Accidental or cheesy bumps are removed;
Shoves and pulls can be used in a teamplay to break gunpoint

## Changelog
:cl:
qol: Gunpoint: Examining the target will show who is holding them at gunpoint
qol: Gunpoint: Examining the shooter will show who they are holding at gunpoint
balance: Gunpoint: If the target tries to grab, they will trigger the shot
balance: Gunpoint: If the target or the shooter are shoved, it will cancel the gunpoint
balance: Gunpoint: If the target is pulled, it will cancel the gunpoint
balance: Both the target and the shooter can't be bumped anymore to avoid cheesing charged shot or removing the gunpoint by just moving around
fix: Clicking the alert button of the shooter will now correctly remove gunpoint
/:cl:

### CODE REVIEW

File: code/datums/components/gunpoint.dm [hunk 1]
```diff
@@ -195,6 +210,28 @@
 		)
 		INVOKE_ASYNC(src, PROC_REF(trigger_reaction))
 
+///Shows if the parent is holding someone at gunpoint
+/datum/component/gunpoint/proc/examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	examine_list += span_warning("<b>[parent] [parent.p_are()] holding [target] at gunpoint with [weapon]!</b>")
```
Comment (MrMelbert) [inline code/datums/components/gunpoint.dm:216]:
```suggestion
	examine_list += span_boldwarning("[parent] [parent.p_are()] holding [target] at gunpoint with [weapon]!")
```

File: code/datums/components/gunpoint.dm [hunk 2]
```diff
@@ -195,6 +210,28 @@
...
+
+///Shows if the examine target is being held at gunpoint
+/datum/component/gunpoint/proc/examine_target(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	examine_list += span_warning("<b>[target] [target.p_are()] being held at gunpoint by [parent]!</b>")
```
Comment (MrMelbert) [inline code/datums/components/gunpoint.dm:221]:
```suggestion
	examine_list += span_boldwarning("[target] [target.p_are()] being held at gunpoint by [parent]!")
```

File: code/datums/status_effects/neutral.dm [hunk 3]
```diff
@@ -152,6 +152,13 @@
 	desc = "You're currently pointing a gun at someone."
```
Comment (Ghommie) [inline code/datums/status_effects/neutral.dm:152]:
```suggestion
	desc = "You're currently pointing a gun at someone. Click to cancel."
```
This lets people know they can cancel the action by clicking the alert.

File: code/datums/components/gunpoint.dm [hunk 4]
```diff
@@ -195,6 +210,28 @@
 		)
 		INVOKE_ASYNC(src, PROC_REF(trigger_reaction))
 
+///Shows if the parent is holding someone at gunpoint
+/datum/component/gunpoint/proc/examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	examine_list += span_boldwarning("[parent] [parent.p_are()] holding [target] at gunpoint with [weapon]!")
```
Comment (Ghommie) [inline code/datums/components/gunpoint.dm:216]: This line shouldn't be here if the viewer can't see the target.

File: code/datums/components/gunpoint.dm [hunk 5]
```diff
@@ -195,6 +210,28 @@
...
+
+///Shows if the examine target is being held at gunpoint
+/datum/component/gunpoint/proc/examine_target(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	examine_list += span_boldwarning("[target] [target.p_are()] being held at gunpoint by [parent]!")
```
Comment (Ghommie) [inline code/datums/components/gunpoint.dm:221]: Viceversa, if the gunman isn't in sight.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/datums/components/gunpoint.dm (hunk 4 above)
Comment (larentoun) [reply code/datums/components/gunpoint.dm:216]: Is `can_see()` proc valid here? It doesn't seem to work correctly on corners, when the shooter is visible on screen, but it's not for the proc. What else is there to use?

Comment (Ghommie) [reply code/datums/components/gunpoint.dm:216]: Yeah, just use `viewers()`. `can_see()` uses a step-towards method / line of sight and it's terrible as the doc itself says. I wonder why we aren't removing it yet.

**Comment by larentoun** [comment]
Now pulling also breaks aim. Probably should've checked if it's forced movement for disarms, pull, etc?

File: code/datums/components/gunpoint.dm [hunk 6]
```diff
@@ -213,12 +213,14 @@
 ///Shows if the parent is holding someone at gunpoint
 /datum/component/gunpoint/proc/examine(datum/source, mob/user, list/examine_list)
 	SIGNAL_HANDLER
-	examine_list += span_boldwarning("[parent] [parent.p_are()] holding [target] at gunpoint with [weapon]!")
+	if(user in viewers(parent))
```
Comment (Ghommie) [inline code/datums/components/gunpoint.dm:216]:
```suggestion
	if(user in viewers(target))
```

Comment (larentoun) [reply code/datums/components/gunpoint.dm]: Oopsie, my bad here

**Pull request merged**